### PR TITLE
no issue: declare var as local

### PR DESCRIPTION
**Pull request by crackevil**

### CODE REVIEW

**Comment by janekptacijarabaci** [comment]
It's the same thing here:
https://github.com/greasemonkey/greasemonkey/blob/3.7/modules/sync.js#L283

**Comment by crackevil** [comment]
Some more commits to merge. Anyone can help to review it for me?Thanks

**Comment by the8472** [comment]
`var` in loop headers is not local to the iteration, only `let` is.

http://stackoverflow.com/questions/18465211/javascript-loop-variable-scope

File: modules/prefmanager.js [hunk 1]
```diff
@@ -127,6 +127,7 @@ GM_PrefManager.prototype.remove = function(prefName) {
  * call a function whenever the named preference subtree changes
  */
 GM_PrefManager.prototype.watch = function(prefName, watcher) {
+  if (this.observers[watcher]) return;
```
Comment (arantius) [inline modules/prefmanager.js]: It's not unreasonable to write a watcher that is capable of observing several preferences, is it not?

**Comment by arantius** [comment]
I've cherry picked 7239e5506aebedf8848139801e64fe91db4bba1a and 3d3b9debc93a179894586e7ea9da074f17bc914b .  Sorry for the delay.

If there's something left which you'd still like to include, please create one new pull request per logical change and it will be considered.

**Pull request closed without merging**